File: backend/app/models/providers.py
from __future__ import annotations
from pydantic import BaseModel


class ProviderCapabilities(BaseModel):
    generation: bool = False
    rigging: bool = False
    animation: bool = False
    textures: bool = False


class ProviderDetail(BaseModel):
    name: str
    display_name: str
    description: str
    stub: bool = False
    enabled: bool = True
    api_key_present: bool = False
    capabilities: ProviderCapabilities = ProviderCapabilities()
    priority_order: int = 99
    health_status: str = "unknown"
    health_message: str = ""


class ProviderTestResult(BaseModel):
    provider: str
    status: str
    message: str
    latency_ms: int | None = None


# Legacy — kept for backwards compat with existing code paths
class ProviderConfig(BaseModel):
    provider_name: str
    enabled: bool = False
    api_key_present: bool = False
    supports_generation: bool = False
    supports_rigging: bool = False
    supports_animation: bool = False
    supports_textures: bool = False
    created_at: str = ""
    updated_at: str = ""


class ProviderJobResult(BaseModel):
    provider: str
    external_job_id: str
    preview_url: str | None = None
    model_url: str | None = None
    thumbnail_url: str | None = None
    raw_response: dict = {}
    completed_at: str | None = None

File: frontend/src/lib/providers.ts
export interface ProviderConfig {
  provider_name: string;
  enabled: boolean;
  api_key_present: boolean;
  supports_generation: boolean;
  supports_rigging: boolean;
  supports_animation: boolean;
  supports_textures: boolean;
  created_at: string;
  updated_at: string;
}

export interface ActiveProvider {
  provider: string;
  is_real: boolean;
}

export async function getProviderStatus(): Promise<ProviderConfig[]> {
  const res = await fetch("/api/providers/status");
  if (!res.ok) throw new Error("Failed to fetch provider status");
  return res.json();
}

export async function getActiveProvider(): Promise<ActiveProvider> {
  const res = await fetch("/api/providers/active");
  if (!res.ok) throw new Error("Failed to fetch active provider");
  return res.json();
}
